File: samples/FsBulletML.Sample.Unity2D.CSharp/Assets/Scripts/ParticleSortingLayer.cs
using UnityEngine;
using System.Collections;

public class ParticleSortingLayer : MonoBehaviour
{
    void Start()
    {
        particleSystem.renderer.sortingLayerName = "Bomb";
        particleSystem.renderer.sortingOrder = 2;
    }

    void Update()
    {
        if (!particleSystem.IsAlive())
        {
            InstanceManager.Destroy(gameObject);
        }
    }
}

File: samples/FsBulletML.Sample.Unity2D.FSharp/FsBulletML.Sample.Unity2D.FSharp/ParticleSortingLayer.fs
namespace FsBulletML.Sample.Unity2D.FSharp

open UnityEngine
open FsBulletML.Unity2D 

type ParticleSortingLayer () = 
  inherit MonoBehaviour ()

  member this.Start () = 
    this.particleSystem.renderer.sortingLayerName <- "Bomb"
    this.particleSystem.renderer.sortingOrder <- 2

  member this.Update () =
    if (this.particleSystem.IsAlive() |> not) then
      InstanceManager.Destroy(this.gameObject)
